### PR TITLE
Enhancements for DiscoGapic message generation

### PR DESCRIPTION
**Pull request by andreamlin**
- Fetch java package name from [api]_gapic.yaml config
- Render copyright and license header
- Give every top-level ViewModel its own TypeTable to manage that file's imports
- Differentiate between a schema's typeName, e.g. "List\<Disk\>", which is how the parent Schema views it, vs the inherent type name (e.g. "Disk") of the schema
- No more arbitrarily nested inner classes; each nested non-primitive-typed JSON schema is now a separate class
- The generated package compiles! (Minus the unfinished [Api]Client.java)

See baseline file diff for resulting changes.

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL

File: src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public String getPathTemplateNameGetter(
   public String getFormatFunctionName(
       Interface apiInterface, SingleResourceNameConfig resourceNameConfig) {
     return publicMethodName(
-        getReducedServiceName(apiInterface.getSimpleName())
+        clientNamePrefix(apiInterface.getSimpleName())
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java]: Why this change?

Comment (andreamlin) [inline src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java]: See above. This file should now match the origin/master version completely.

File: src/test/java/com/google/api/codegen/testdata/go_main_library.baseline [hunk 2]
```diff
@@ -211,8 +211,8 @@ func (c *Client) SetGoogleClientInfo(keyval ...string) {
     c.xGoogHeader = []string{gax.XGoogHeader(kv...)}
 }
 
-// LibraryShelfPath returns the path for the shelf resource.
-func LibraryShelfPath(shelfId string) string {
+// ShelfPath returns the path for the shelf resource.
```
Comment (andreamlin) [inline src/test/java/com/google/api/codegen/testdata/go_main_library.baseline]: Right before this PR, I merged the newest origin/master into origin/discogapic and for whatever reason, some golang files missed out on the update. All the changes to golang files in this PR are just to bring them in sync (exact copies) of origin/master.

File: src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSchemaToViewTransformer.java [hunk 3]
```diff
@@ -88,23 +96,28 @@ public JavaDiscoGapicSchemaToViewTransformer(
...
             JavaFeatureConfig.newBuilder().enableStringFormatFunctions(false).build());
 
-    StaticLangApiMessageFileView apiFile;
-
-    // TODO(andrealin): Remove this when imports are fixed.
-    final List<Schema> schemas = new LinkedList<>(context.getDocument().schemas().values());
+    // Escape any schema's field names that are Java keywords.
+    Set<String> reservedKeywords = new HashSet<>();
+    reservedKeywords.addAll(JavaNameFormatter.RESERVED_IDENTIFIER_SET);
+    reservedKeywords.add("Builder");
```
Comment (andreamlin) [inline src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSchemaToViewTransformer.java]: Removed the local variable.

File: src/test/java/com/google/api/codegen/testdata/go_main_library.baseline (hunk 2 above)
Comment (andreamlin) [inline src/test/java/com/google/api/codegen/testdata/go_main_library.baseline]: I've reviewed the entire diff of this PR against origin/master, and all the changes should be relevant to Discovery gen now.

**Comment by andreamlin** [comment]
Responded to comments and removed redundancies. PTAL

**Comment by andreamlin** [comment]
Blocked on https://github.com/googleapis/toolkit/pull/1373.

**Comment by andreamlin** [comment]
Removed the golang diffs in a prior PR. PTAL.

**Comment by garrettjonesgoogle** [comment]
LGTM
